### PR TITLE
[WIP] performance mvp

### PR DESCRIPTION
**Pull request by yzhang1994**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [ ] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

- Related Clubhouse Story: https://app.clubhouse.io/coinalpha/story/3041/performance-monitoring-mvp

**A description of the changes proposed in the pull request**:
- Stripped out the trade listener on the StrategyBase class (It is impossible to figure out the source of the event this way, therefore using EventLogger attached to MarketBase instead)
- Refactored existing EventLogger to cython, and adding an event source (It shouldn't break any existing code)
- Extended Trade class to also store OrderType, market_name, and a timestamp
- Adding `performance` function that gives a snapshot of past trades and balance changes
image
- Adding `export_trades` function that would export past trades to a csv file 

@martinkou Please take a look at the changes and see if this approach makes sense.

### CODE REVIEW

**Comment by mifeng** [comment]
Let's call the command `history` rather than `performance`: it's shorter and more reflective of what the command does today.

**Comment by mifeng** [comment]
1) I suggest a different tabular format for balances:

Balances:
| Market | Asset | Starting | Current |
------------------------------------
| Binance | ETH | value | value |
| binance | ZRX | value | value |
| binance | ETH | value | value |
| radar_relay | ZRX | value | value |
| radar_relay | WETH | 0.1 | 0.1 |

2) The name of the market should be `binance` rather than `BinanceMarket` for consistency

3) Format of timestamp is a bit strange. Ideally it's `2019-04-24 16:55:56`. This should be same time zone as log message timestamps.

4) (this is optional and not that important) You can combine the `trade_side` and the `order_type` fields by using `LIMIT BUY` or `MARKET SELL`. 

5) Is price net of fees and gas?

**Comment by yzhang1994** [comment]
@mifeng 
1. Can change the table format
2. This is actually difficult to do unless I can add a market_name property to each market. @martinkou  Can I do that?
3. Can change timestamp format easily
4. Can combine order_side and order_type
5. The price is not net of fees and gas. I'm waiting on @chowryan's gas/fees PR to be merged. After that, I probably will have to add fees and gas to the OrderFilledEvent in each of the exchange connectors, which will take a while to do. Will need to figure out the details later.

**Comment by martinkou** [comment]
Yes, feel free to add a `name` property to market base.

File: wings/market/market_base.pyx [hunk 1]
```diff
@@ -30,8 +31,14 @@ cdef class MarketBase(TimeIterator):
     def __init__(self):
         super().__init__()
         self.event_reporter = EventReporter(event_source=self.__class__.__name__)
+        self.event_logger = EventLogger(event_source=self.__class__.__name__)
         for event_tag in self.MARKET_EVENTS:
             self.c_add_listener(event_tag.value, self.event_reporter)
+            self.c_add_listener(event_tag.value, self.event_logger)
+
+    @property
+    def event_logs(self) -> Dict[str, OrderBook]:
```
Comment (martinkou) [inline wings/market/market_base.pyx]: the return type is wrong.

Comment (yzhang1994) [reply wings/market/market_base.pyx]: fixed

**Pull request closed without merging**